### PR TITLE
Arrays.copy vs copyOf vs System.arrayCopy

### PR DESCRIPTION
**Pull request by dominikjaniga91**
I added additional comparison for copy with Stream and SerializationUtils

### CODE REVIEW

**Comment by marcin-ogorzalek-EPAM** [comment]
Please increase the project's version in the `pom.xml`

File: src/main/java/com/epam/prejap/ess/helpers/ArraysCopy.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
+package com.epam.prejap.ess.helpers;
```
Comment (marcin-ogorzalek-EPAM) [inline src/main/java/com/epam/prejap/ess/helpers/ArraysCopy.java:1]: Please enclose your solution in an additional package (inside the main one) with a name indicating the solved problem. Don't use the main application's package

Comment (dominikjaniga91) [reply src/main/java/com/epam/prejap/ess/helpers/ArraysCopy.java]: fixed:)

File: src/main/java/com/epam/prejap/ess/helpers/arrays/copy/ArraysCopy.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+ * @version 0.3
+ */
+@Fork(value = 1)
+@Threads(value = 1)
+@BenchmarkMode(Mode.AverageTime)
+@Measurement(iterations = 100, timeUnit = TimeUnit.MILLISECONDS,  time = 10)
+@Warmup(iterations = 10, timeUnit = TimeUnit.MILLISECONDS, time = 10)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@Timeout(time = 1, timeUnit = TimeUnit.MICROSECONDS)
+public class ArraysCopy {
```
Comment (marcin-ogorzalek-EPAM) [inline src/main/java/com/epam/prejap/ess/helpers/arrays/copy/ArraysCopy.java:29]: Please tell me, if this class and some methods in it have to be `public` for JMH or their visibility can be lowered?

Comment (dominikjaniga91) [reply src/main/java/com/epam/prejap/ess/helpers/arrays/copy/ArraysCopy.java:29]: JMH generates its own classes in directory target/generated-sources/annotations. Without public modifier it does not work

image

Method with Benchmark annotations also has to be public

image

**Pull request merged**